doc/MGMT_CLI.md

```diff
@@ -0,0 +1,420 @@
+
+
+#### Telemetry Server
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/port
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/client_auth
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/jwt_refresh
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/jwt_valid
+
+/restconf/data/sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST=gnmi/enable
+
+
+### 3.6.4 gNMI Support
+
+gNMI Will support the same paths as REST but using the gNMI path/key format, i.e:
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/port
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/client_auth
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/jwt_refresh
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/jwt_valid
+
+sonic-management:sonic-management/REST_SERVER/REST_SERVER_LIST[server=default]/enable
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/port
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/client_auth
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/jwt_refresh
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/jwt_valid
+
+sonic-management:sonic-management/TELEMETRY/TELEMETRY_LIST[server=gnmi]/enable
+
+
+
+## 3.7 Warm Boot Support
+N/A
+
+## 3.8 Upgrade and Downgrade Considerations
+
+The client_auth field will need to be migrated from the old format of a string with comma separated values to the new enum based field. A migrator function will be added to the migration script to handle this.
+
+The existing fields are a string with comma separated values (CSV). Since these values can be in any order, and the "none" option can invalidate the other modes, this must be considered during the migration.
+
+Below is a list of valid migrations:
+
+- password -> PASSWORD
+- password,jwt -> PASSWORD_JWT
+- none -> NONE
+- password,jwt,cert -> PASSWORD_JWT_CERT
+- password,cert -> PASSWORD_CERT
+- jwt,cert -> JWT_CERT
+- cert -> CERT
+
+
+
+## 3.9 Resource Needs
+
+No significant increases in resources needed.
+
+# 4 Flow Diagrams
+N/A
+
+# 5 Error Handling
+
+The model should prevent invalid configurations from being created, such as invalid client_auth modes (like JWT alone) which is handled by using an ENUM of valid modes. For PORT field, valid ports are between 1-65535, 0 should not be allowed and should result in an error. Both jwt_refresh and jwt_valid should not allow 0 either. This is acheived by the YANG range statement. In the case an invalid value is specified on the CLI, the CLI will display the error.
+
+In the case that the user specifies a port that is already in use, this will cause the service to not restart successfully. In this case the service should revert back to the default port.
```

Comment:
This may not be a good idea. If admin configures a port, he will not be expecting the service to come up with default port. Next restart may use the configured port depending on the system state. Clients/controllers cannot change their settings dynamically.